### PR TITLE
add print support for nexys. Add print to Sconstruct for all the scon…

### PR DESCRIPTION
**Pull request by oferShinaar**
…scripts that build in the "build session"

### CODE REVIEW

**Review by nati-rapaport — APPROVED** [review]

File: WD-Firmware/.project [hunk 1]
```diff
@@ -5,6 +5,11 @@
 	<projects>
 	</projects>
 	<buildSpec>
+		<buildCommand>
```
Comment (ronen-haen-sndk) [inline WD-Firmware/.project:8]: I think you should revert the changes in the .project

File: WD-Firmware/board/nexys_a7_swerv1/bsp/mem_map.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+#ifndef __MEM_MAP_H
+#define __MEM_MAP_H
+
+//Memory map
+#define EXT_SRAM_BASE_ADDRES 	0x00000000
+#define ICCM_BASE_ADDRESS		0xEE000000
+#define DCCM_BASE_ADDRESS		0xF0080000
+#define UART_BASE_ADDRESS		0x44A00000
```
Comment (ronen-haen-sndk) [inline WD-Firmware/board/nexys_a7_swerv1/bsp/mem_map.h:24]: #defines should start with D_

File: WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+/*---------------------------------------------------*/
+/* Define											 */ 
+/*---------------------------------------------------*/
+#define UART_RX_DATA	(*((volatile unsigned int*)(UART_BASE_ADDRESS + 0x0)))
+#define UART_TX_DATA	(*((volatile unsigned int*)(UART_BASE_ADDRESS + 0x4)))
+#define UART_STAT		(*((volatile unsigned int*)(UART_BASE_ADDRESS + 0x8)))
+#define UART_CTRL 		(*((volatile unsigned int*)(UART_BASE_ADDRESS + 0xC)))
+
+#define UART_TX_BUSY	(1<<3)
+#define UART_RX_AVAIL	(1<<0)
```
Comment (ronen-haen-sndk) [inline WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c:36]: defines should start w/ D_

File: WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+
+/*---------------------------------------------------*/
+/* Define											 */ 
+/*---------------------------------------------------*/
+#define UART_RX_DATA	(*((volatile unsigned int*)(UART_BASE_ADDRESS + 0x0)))
+#define UART_TX_DATA	(*((volatile unsigned int*)(UART_BASE_ADDRESS + 0x4)))
+#define UART_STAT		(*((volatile unsigned int*)(UART_BASE_ADDRESS + 0x8)))
+#define UART_CTRL 		(*((volatile unsigned int*)(UART_BASE_ADDRESS + 0xC)))
+
+#define UART_TX_BUSY	(1<<3)
```
Comment (ronen-haen-sndk) [inline WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c:35]: Should we add a define for the 1 and 3 for code to be more readable? same goes for UART_RX_AVAIL

File: WD-Firmware/demo/build/SConstruct [hunk 5]
```diff
@@ -132,14 +137,20 @@ Env['BUILDERS']['Size'] = Builder(action = utils.fnProduceSectionsSize)
 Env['PUBLIC_DEF'] = objDemo.public_defs
 Env['TARGET_LIBS'] += ['c', 'gcc']
 
+#prepare all Sconscripts for build
+objDemo.listSconscripts.append(strBsp+ Env['TARGET_BOARD']) 
+
 Env["Scan"] = True
-SConscript('SConscript_bsp_'+ Env['TARGET_BOARD'], exports='Env')
 for strSconscript in objDemo.listSconscripts:
   SConscript('SConscript_%s' % strSconscript, exports='Env')
```
Comment (nidalf) [inline WD-Firmware/demo/build/SConstruct]: make a string variable at the beginning of the file
strSconscriptPrefix = "SConscript_%s"

File: WD-Firmware/board/nexys_a7_swerv1/bsp/mem_map.h (hunk 2 above)
Comment (oferShinaar) [reply WD-Firmware/board/nexys_a7_swerv1/bsp/mem_map.h:24]: Correct, but not on this round.
this file receive from Arup repo. once we will see it works we will also fix syntax over the all file

File: WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c (hunk 4 above)
Comment (oferShinaar) [reply WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c:35]: same as above

File: WD-Firmware/demo/build/SConstruct (hunk 5 above)
Comment (oferShinaar) [reply WD-Firmware/demo/build/SConstruct]: done but without the %s (have to be const)
strSconscriptPrefix = 'SConscript_'

File: WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+typedef struct params_s 
+{
+    int len;
+    long num1;
+    long num2;
+    char pad_character;
+    int do_padding;
+    int left_flag;
+	int upper_hex_digit_flag; //added hexdigit uppercase [A-F]
+	int maxium_length; // max_length
```
Comment (ronen-haen-sndk) [inline WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c:83]: shouldn't we use our types?

File: WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+    UART_TX_DATA = ch;
+
+    return 0;
+}
+
+/*----------------------------------------------------*/
+/* Use the following parameter passing structure to   */
+/* make printf re-entrant.                         */
+/*----------------------------------------------------*/
+typedef struct params_s
```
Comment (ronen-haen-sndk) [inline WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c:74]: structure not following the coding convention

File: WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c [hunk 8]
```diff
@@ -0,0 +1,396 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//
+// Based on a public domain version of printf
+// Web link: http://www.rte.se/sites/default/files/Blog/Modesty/ee_printf.c 
+//
+
+
```
Comment (ronen-haen-sndk) [inline WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c:21]: general comment - the file implementation doesn't follow the coding convention

File: WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c (hunk 6 above)
Comment (oferShinaar) [reply WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c:83]: yes, not in this phase

File: WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c (hunk 7 above)
Comment (oferShinaar) [reply WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c:74]: yes, not in this phase

File: WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c (hunk 8 above)
Comment (oferShinaar) [reply WD-Firmware/board/nexys_a7_swerv1/bsp/printf.c:21]: yes, not in this phase

**Review by ronen-haen-sndk — APPROVED** [review]
